tiatoolbox/tools/registration/wsi_registration.py

```diff
@@ -190,8 +191,81 @@ def match_histograms(
     return image_a, image_b
 
 
+class DFBRFeatureExtractor(torch.nn.Module):
+    """Feature extractor for Deep Feature based Registration (DFBR).
+
+    This class extracts features from three different layers of VGG16.
+    These features are processed in DFBRegister class for registration
+    of a pair of images.
+
+    """
+
+    def __init__(self):
+        super().__init__()
+        output_layers_id = ["16", "23", "30"]
+        output_layers_key = ["block3_pool", "block4_pool", "block5_pool"]
+        self.features = OrderedDict.fromkeys(output_layers_key, None)
+        self.pretrained = torchvision.models.vgg16(pretrained=True).features
+        self.f_hooks = []
+
+        for i, l in enumerate(output_layers_id):
+            self.f_hooks.append(
+                getattr(self.pretrained, l).register_forward_hook(
+                    self.forward_hook(output_layers_key[i])
+                )
+            )
+
+    def forward_hook(self, layer_name):
+        """Register a hook.
+
+        Args:
+            layer_name (str):
+                User-defined name for a layer.
+
+        Returns:
+            None
+
+        """
+
+        def hook(_0, _1, module_output):
+            """Forward hook.
+
+            Args:
+                _0:
+                    Unused argument for the module.
+                _1:
+                    Unused argument for the modules' input.
+                module_output (torch.Tensor):
+                    Output (features) of the module.
+
```

Comment:
Could these both just be `_` or would the linters complain?

Reply:
https://github.com/TissueImageAnalytics/tiatoolbox/blob/6eeaea529576f4b51b2c90530f30154343b3ce45/tiatoolbox/tools/registration/wsi_registration.py#L232-L246[](url)